ui/src/adapters/parsers/jsonSchemas.ts

```diff
@@ -67,96 +114,117 @@ const numberSchemaParser = StrictSchema<domain.NumberSchema>()(
   )
 );
 
-const numberAttributeParser = (name: string, required: boolean) =>
-  StrictSchema<domain.NumberSchema, domain.NumberAttribute>()(
+function getNumberAttributeParser(name: string, required: boolean) {
+  return getStrictParser<NumberSchema, NumberAttribute>()(
     numberSchemaParser.transform(
-      (schema): domain.NumberAttribute => ({
+      (schema): NumberAttribute => ({
         name,
         required,
         schema,
         type: "number",
       })
     )
   );
+}
 
-const booleanPropsParser = StrictSchema<domain.BooleanProps>()(
+const stringPropsParser = getStrictParser<StringProps>()(
   z.object({
-    enum: z.boolean().array().min(1).optional(),
+    enum: z.string().array().min(1).optional(),
+    format: z.string().optional(),
   })
 );
 
-const booleanSchemaParser = StrictSchema<domain.BooleanSchema>()(
-  commonPropsParser.and(booleanPropsParser).and(
+const stringSchemaParser = getStrictParser<StringSchema>()(
+  commonPropsParser.and(stringPropsParser).and(
     z.object({
-      type: z.literal("boolean"),
+      type: z.literal("string"),
     })
   )
 );
 
-const booleanAttributeParser = (name: string, required: boolean) =>
-  StrictSchema<domain.BooleanSchema, domain.BooleanAttribute>()(
-    booleanSchemaParser.transform(
-      (schema): domain.BooleanAttribute => ({
+function getStringAttributeParser(name: string, required: boolean) {
+  return getStrictParser<StringSchema, StringAttribute>()(
+    stringSchemaParser.transform(
+      (schema): StringAttribute => ({
         name,
         required,
         schema,
-        type: "boolean",
+        type: "string",
       })
     )
   );
+}
 
-const nullSchemaParser = StrictSchema<domain.NullSchema>()(
-  commonPropsParser.and(
+// Non-primitive
+
+type ArrayPropsInput = {
+  item?: unknown;
+};
+
+type ArraySchemaInput = CommonProps & ArrayPropsInput & { type: "array" };
+
+const arrayPropsParser = getStrictParser<ArrayPropsInput>()(
```

Comment:
Now that these types have the `Input` prefix, their parsers should have it also. The rule is that the name of a parser is formed with the name of the type + the suffix `Parser`.